### DISCUSSION TITLE
Rebuilding old versions?

### DISCUSSION THREAD

**Discussion by LouiseWraaHansen** [actions/virtual-environments#9954] [General questions]
The Windows Server 2022 build currently has a latest success dating back to 20240514.
I would like to use the corresponding tag/commit id to build a a customized image with a few more tools but if I try to build this version I get issues with thumbprints:

    azure-arm.image: Install chocolatey
    azure-arm.image: Downloading package from https://chocolatey.org/install.ps1 to C:\Users\packer\AppData\Local\Temp\install.ps1...
    azure-arm.image: Package downloaded in 0.36 seconds
==> azure-arm.image: Signature thumbprint do not match expected.
==> azure-arm.image: At C:\Program Files\WindowsPowerShell\Modules\ImageHelpers\InstallHelpers.ps1:972 char:9

I suppose this in general is due to installing 'latest' version of this or that but with a fixed thumbprint(?) so is it possible to build older versions like this at all?

/Louise

**Comment by mikhailkoliada** [comment]
Hello! We do not support building from the release branches, please always use main for your custom scripts as described here: https://github.com/actions/runner-images/issues/9884, I can confirm that from the main branch builds now are functioning

**Reply by LouiseWraaHansen** [reply]
Hi, thanks for your reply but I am not sure I understand it as an answer to my question.

As you are probably aware the windows 2022 variant had not been successfully built for a couple of weeks at the time of my question.
Thus I tried to build main but with the commit id identified by latest successful build in the section 'Available images' on the main page which was from 20240513 or something similar.
I am not a regular github user so excuse me if I am missing something obvious.

Another way to rephrase my original question would be: what can I do if I want to build say Windows 2022 and the latest pushes to main make it fail for whatever reason? 
Is my only option to keep an eye out for a successful build and start one myself when so hoping nobody has pushed something bad inbetween?

/Louise
